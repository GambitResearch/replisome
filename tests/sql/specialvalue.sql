\set VERBOSITY terse
\pset format unaligned

-- predictability
SET synchronous_commit = on;

DROP TABLE IF EXISTS xpto;

SELECT slot_create();

CREATE TABLE xpto (a SERIAL PRIMARY KEY, b bool, c varchar(60), d real);
COMMIT;

BEGIN;
INSERT INTO xpto (b, c, d) VALUES('t', 'test1', '+inf');
INSERT INTO xpto (b, c, d) VALUES('f', 'test2', 'nan');
INSERT INTO xpto (b, c, d) VALUES(NULL, 'null', '-inf');
INSERT INTO xpto (b, c, d) VALUES(TRUE, E'valid: '' " \\ / \b \f \n \r \t \u207F \u967F invalid: \\g \\k end', 123.456);
COMMIT;

SELECT data FROM slot_get();
SELECT slot_drop();
