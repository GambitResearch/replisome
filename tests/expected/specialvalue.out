\set VERBOSITY terse
\pset format unaligned
-- predictability
SET synchronous_commit = on;
DROP TABLE IF EXISTS xpto;
SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'replisome');
?column?
init
(1 row)
CREATE TABLE xpto (a SERIAL PRIMARY KEY, b bool, c varchar(60), d real);
COMMIT;
WARNING:  there is no transaction in progress
BEGIN;
INSERT INTO xpto (b, c, d) VALUES('t', 'test1', '+inf');
INSERT INTO xpto (b, c, d) VALUES('f', 'test2', 'nan');
INSERT INTO xpto (b, c, d) VALUES(NULL, 'null', '-inf');
INSERT INTO xpto (b, c, d) VALUES(TRUE, E'valid: '' " \\ / \b \f \n \r \t \u207F \u967F invalid: \\g \\k end', 123.456);
COMMIT;
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'pretty-print', '1');
data
{
	"change": [
		{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"colnames": ["a", "b", "c", "d"],
			"coltypes": ["int4", "bool", "varchar", "float4"],
			"values": [1, true, "test1", null]
		}
		,{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [2, false, "test2", null]
		}
		,{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [3, null, "null", null]
		}
		,{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [4, true, "valid: ' \" \\ \/ \b \f \n \r \t ⁿ 陿 invalid: \\g \\k end", 123.456]
		}
	]
}
(1 row)
SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
?column?
stop
(1 row)
